packages/astro/client.d.ts

```diff
@@ -0,0 +1,207 @@
+/// <reference types="vite/types/importMeta" />
+
+// CSS modules
+type CSSModuleClasses = { readonly [key: string]: string }
+
+declare module '*.module.css' {
+  const classes: CSSModuleClasses
+  export default classes
+}
+declare module '*.module.scss' {
+  const classes: CSSModuleClasses
+  export default classes
+}
+declare module '*.module.sass' {
+  const classes: CSSModuleClasses
+  export default classes
+}
+declare module '*.module.less' {
+  const classes: CSSModuleClasses
+  export default classes
+}
+declare module '*.module.styl' {
+  const classes: CSSModuleClasses
+  export default classes
+}
+declare module '*.module.stylus' {
+  const classes: CSSModuleClasses
+  export default classes
+}
+declare module '*.module.pcss' {
+  const classes: CSSModuleClasses
+  export default classes
+}
+
+// CSS
+declare module '*.css' {
+  const css: string
+  export default css
+}
+declare module '*.scss' {
+  const css: string
+  export default css
+}
+declare module '*.sass' {
+  const css: string
+  export default css
+}
+declare module '*.less' {
+  const css: string
+  export default css
+}
+declare module '*.styl' {
+  const css: string
+  export default css
+}
+declare module '*.stylus' {
+  const css: string
+  export default css
+}
+declare module '*.pcss' {
+  const css: string
+  export default css
+}
+
+// Built-in asset types
+// see `src/constants.ts`
+
+// images
+declare module '*.jpg' {
+  const src: string
+  export default src
+}
+declare module '*.jpeg' {
+  const src: string
+  export default src
+}
+declare module '*.png' {
+  const src: string
+  export default src
+}
```

Comment:
I think this means @tony-sull can override these now?

Reply:
By override do you mean change them, then sure. If you mean override in his project, I'm not enough of a TS guru to know.

Reply:
I think typescript will let `@astrojs/image` re-declare these, but I'm not 100% sure.   @matthewp the main need is for `@astrojs/image` to be able to define all image imports to something similar to 

```ts
declare module '*.jpeg' {
  const metadata: {
    src: string
    width: number
    height: number
    format: string
  }
  export default metadata
}
```

Reply:
I believe adding these to a `@astrojs/image/client.d.ts` file and then adding that to `compilerOptions.include` will override these. We don't want this to reference the `@astrojs/image` types unless `@astrojs/image` is installed.